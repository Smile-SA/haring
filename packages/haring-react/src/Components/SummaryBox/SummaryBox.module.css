.grow {
  flex-grow: 1;
}

.titleNode {
  font-weight: 700;
}

.topNode {
  color: '#5C5F66';
  font-size: 12px;
}
