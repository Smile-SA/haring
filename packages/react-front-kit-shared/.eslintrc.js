module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
