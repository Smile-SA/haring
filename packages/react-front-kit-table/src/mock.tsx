/* eslint-disable react-refresh/only-export-components */
export * from './Components/Table/Table.mock';
export * from './Components/TableGridView/TableGridView.mock';
