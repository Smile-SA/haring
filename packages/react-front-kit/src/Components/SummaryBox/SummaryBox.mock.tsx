import type { ReactElement } from 'react';

import {
  ActionIcon,
  Avatar,
  Button,
  Flex,
  Indicator,
  Text,
} from '@mantine/core';
import { Export, Eye, UserCircle } from '@phosphor-icons/react';

import { ActionRowOverflow } from '../ActionRowOverflow/ActionRowOverflow';
import {
  actionRowOverflowActionsMock,
  actionRowOverflowSelectedMock,
} from '../ActionRowOverflow/ActionRowOverflow.mock';

export const defaultSummaryBoxMock = {
  children: 'children node',
  leftNode: 'left node',
  rightNode: 'right node',
  titleNode: 'title node',
  topNode: 'top node',
};

export const topMock = (
  <span style={{ color: '#5C5F66', fontSize: 12 }}>30th novembre 2023</span>
);

export const otherTopMock = (
  <span style={{ color: '#5C5F66', fontSize: 12 }}>3 hours ago</span>
);

export const titleMock = <b>Lorem ipsum dolor sit amet</b>;

export const otherTitleMock = (
  <span>
    <b>Lorem ipsum</b>
    <span>
      dolor sit amet consectetur. Sollicitudin mattis blandit aliquet odio urna
      mi id :
    </span>
    <Text c="blue" component="span">
      Lorem Ipsum dolor
    </Text>
  </span>
);

export const childrenExampleMock = (key?: number): ReactElement => (
  <Flex key={key} direction="column" gap={24}>
    <span>
      Lorem ipsum dolor sit amet consectetur. Sollicitudin mattis blandit
      aliquet odio urna mi id. Mauris venenatis ut et at amet vel est.
    </span>
    <Flex justify="space-between" style={{ width: '100%' }}>
      <Button color="green" variant="light">
        Approved
      </Button>
      <Flex align="center" gap={16}>
        <ActionIcon color="cyan" size={30} variant="subtle">
          <Export size={24} />
        </ActionIcon>
        <ActionIcon color="cyan" size={30} variant="subtle">
          <Eye size={24} />
        </ActionIcon>
      </Flex>
    </Flex>
  </Flex>
);

export const otherChildrenExampleMock = (
  <Flex direction="column" gap={24}>
    <span>
      Lorem ipsum dolor sit amet consectetur. Sollicitudin mattis blandit
      aliquet odio urna mi id.
    </span>
    <Flex gap={5} justify="start" style={{ width: '100%' }} wrap="wrap">
      <Button color="green" variant="light">
        Approved
      </Button>
      <Button color="orange" variant="light">
        In Progress
      </Button>
      <Button color="white" variant="light">
        Paused
      </Button>
      <Button color="red" variant="light">
        Blocked
      </Button>
      <Button color="black" variant="default">
        Rejected
      </Button>
    </Flex>
  </Flex>
);

export const rightMock = (
  <ActionIcon color="cyan" size={30} variant="subtle">
    <Export size={24} />
  </ActionIcon>
);

export const actionRightMock = (
  <ActionRowOverflow
    actions={actionRowOverflowActionsMock}
    isCompactStyle
    rowActionNumber={1}
    selectedElements={actionRowOverflowSelectedMock}
  />
);

export const leftMock = (
  <Indicator color="red" inline offset={8} withBorder>
    <Avatar size={40}>
      <UserCircle size={40} />
    </Avatar>
  </Indicator>
);
