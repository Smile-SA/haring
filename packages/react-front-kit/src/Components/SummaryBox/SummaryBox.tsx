'use client';

import type { FlexProps, GroupProps, StackProps } from '@mantine/core';
import type { ReactElement, ReactNode } from 'react';

import { Flex, Group, Stack } from '@mantine/core';

import classes from './SummaryBox.module.css';

export interface ISummaryBoxProps extends FlexProps {
  children?: ReactNode;
  childrenGroupProps?: GroupProps;
  contentContainerFlexProps?: FlexProps;
  leftNode?: ReactNode;
  leftNodeStackProps?: StackProps;
  rightNode?: ReactNode;
  rightNodeStackProps?: StackProps;
  titleNode?: ReactNode;
  titleNodeGroupProps?: GroupProps;
  topNode?: ReactNode;
  topNodeGroupProps?: GroupProps;
}

export function SummaryBox(props: ISummaryBoxProps): ReactElement {
  const {
    children,
    childrenGroupProps,
    contentContainerFlexProps,
    leftNode,
    leftNodeStackProps,
    rightNode,
    rightNodeStackProps,
    titleNode,
    titleNodeGroupProps,
    topNode,
    topNodeGroupProps,
    ...containerFlexProps
  } = props;

  return (
    <Flex
      align="stretch"
      gap="md"
      justify="stretch"
      wrap="nowrap"
      {...containerFlexProps}
    >
      {Boolean(leftNode) && (
        <Stack align="stretch" justify="flex-start" {...leftNodeStackProps}>
          {leftNode}
        </Stack>
      )}
      {Boolean(topNode || titleNode || children) && (
        <Flex
          align="stretch"
          className={classes.grow}
          direction="column"
          gap={4}
          justify="start"
          {...contentContainerFlexProps}
        >
          {Boolean(topNode) && <Group {...topNodeGroupProps}>{topNode}</Group>}
          {Boolean(titleNode) && (
            <Group {...titleNodeGroupProps}>{titleNode}</Group>
          )}
          {Boolean(children) && (
            <Group className={classes.grow} {...childrenGroupProps}>
              {children}
            </Group>
          )}
        </Flex>
      )}
      {Boolean(rightNode) && (
        <Stack align="stretch" justify="flex-start" {...rightNodeStackProps}>
          {rightNode}
        </Stack>
      )}
    </Flex>
  );
}
