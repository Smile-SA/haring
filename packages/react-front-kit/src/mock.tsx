/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
