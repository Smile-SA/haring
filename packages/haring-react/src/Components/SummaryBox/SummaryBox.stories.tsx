import type { Meta, StoryObj } from '@storybook/react';

import { Container } from '@mantine/core';

import { SummaryBox as Cmp } from './SummaryBox';
import {
  actionRightMock,
  childrenExampleMock,
  defaultSummaryBoxMock,
  leftMock,
  otherChildrenExampleMock,
  otherChildrenMock,
  otherTopMock,
  rightMock,
  titleMock,
  topMock,
} from './SummaryBox.mock';

const meta = {
  component: Cmp,
  decorators: [
    function Component(Story, ctx) {
      return (
        <Container bg="white" p="1rem" size="400px">
          <Story args={{ ...ctx.args }} />
        </Container>
      );
    },
  ],
  tags: ['autodocs'],
  title: '3-custom/Components/SummaryBox',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const SummaryBox: IStory = {
  args: defaultSummaryBoxMock,
};

export const ChildrenExample: IStory = {
  args: {
    children: childrenExampleMock(),
    titleNode: titleMock,
    topNode: topMock,
    topNodeGroupProps: { mb: 12 },
  },
};

export const SimpleRightNodeExample: IStory = {
  args: {
    pb: 24,
    rightNode: rightMock,
    titleNode: titleMock,
  },
};

export const ActionRightNodeExample: IStory = {
  args: {
    children: otherChildrenExampleMock,
    rightNode: actionRightMock,
    rightNodeStackProps: { justify: 'center', style: { minWidth: 70 } },
    titleNode: titleMock,
    topNode: topMock,
    topNodeGroupProps: { mb: 12 },
  },
};

export const LeftNodeExample: IStory = {
  args: {
    children: otherChildrenMock,
    leftNode: leftMock,
    topNode: otherTopMock,
  },
};
