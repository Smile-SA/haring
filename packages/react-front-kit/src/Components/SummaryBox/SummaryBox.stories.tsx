import type { Meta, StoryObj } from '@storybook/react';
import type { ReactElement } from 'react';

import { Container } from '@mantine/core';

import { SummaryBox as Cmp } from './SummaryBox';
import {
  childrenExampleMock,
  defaultSummaryBoxMock,
  leftMock,
  otherTitleMock,
  otherTopMock,
  rightMock,
  titleMock,
  topMock,
} from './SummaryBox.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/SummaryBox',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const SummaryBox: IStory = {
  args: defaultSummaryBoxMock,
};

const container = ({ ...props }): ReactElement => (
  <Container bg="white" size="400px">
    <Cmp {...props} />
  </Container>
);

export const ChildrenExample: IStory = {
  args: {
    children: childrenExampleMock,
    titleNode: titleMock,
    topNode: topMock,
    topNodeGroupProps: { mb: 12 },
  },
  render: ({ ...props }) => container(props),
};

export const SimpleRightNodeExample: IStory = {
  args: {
    pb: 24,
    rightNode: rightMock,
    titleNode: titleMock,
  },
  render: ({ ...props }) => container(props),
};

export const LeftNodeExample: IStory = {
  args: {
    leftNode: leftMock,
    titleNode: otherTitleMock,
    topNode: otherTopMock,
  },
  render: ({ ...props }) => container(props),
};
