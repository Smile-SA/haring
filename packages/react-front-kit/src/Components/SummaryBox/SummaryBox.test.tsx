import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { SummaryBox } from './SummaryBox';
import { defaultSummaryBoxMock } from './SummaryBox.mock';

describe('SummaryBox', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <SummaryBox id="summary-card" {...defaultSummaryBoxMock} />,
    );
    expect(container).toMatchSnapshot();
  });
});
