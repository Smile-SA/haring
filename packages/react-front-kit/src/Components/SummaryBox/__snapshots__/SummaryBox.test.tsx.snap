// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryBox matches snapshot 1`] = `
<div>
  <div
    class="mantine-1nzdwls"
    id="summary-card"
  >
    <div
      class="mantine-Stack-root mantine-caxjnw"
    >
      left node
    </div>
    <div
      class="mantine-1ruz0vj"
    >
      <div
        class="mantine-Group-root mantine-k3ov3c"
      >
        top node
      </div>
      <div
        class="mantine-Group-root mantine-k3ov3c"
      >
        title node
      </div>
      <div
        class="mantine-Group-root mantine-w3lnwg"
      >
        children node
      </div>
    </div>
    <div
      class="mantine-Stack-root mantine-caxjnw"
    >
      right node
    </div>
  </div>
</div>
`;
